nixpkgs_merge_bot/github/github_client.py

```diff
@@ -231,23 +231,33 @@ def create_issue_reaction(
         )
 
     def merge_pull_request(
-        self, owner: str, repo: str, pr_number: int, sha: str, commenter: dict[str, Any]
+        self, pr_number: int, node_id: str, sha: str
     ) -> HttpResponse | None:
         if STAGING:
             log.debug(f"pull request {pr_number}: Staging, not merging")
             return None
 
-        committer_username = commenter["login"]
-        committer_email = (
-            commenter["email"]
-            if commenter["email"]
-            else f"{committer_username}@users.noreply.github.com"
-        )
-        commit_message = f"Co-authored-by: {committer_username} <{committer_email}>"
-        log.debug(f"pull request {pr_number}, commit_message: {commit_message}")
-        return self.put(
-            f"/repos/{owner}/{repo}/pulls/{pr_number}/merge",
-            data={"sha": sha, "commit_message": commit_message},
+        # Using GraphQL's enablePullRequestAutoMerge mutation instead of the REST
+        # /merge endpoint, because the latter doesn't work with Merge Queues.
+        # This mutation works both with and without Merge Queues, and independently
+        # of whether the PR's required checks are still pending or have already
+        # succeeded.
+        # The only case where it doesn't work, is when there are no required status
+        # checks for the target branch. All development branches have these enabled,
```

Comment:
Question just for my own curiosity and piece of mind:

Would it still work on a repo/branch that doesn't allow "Auto Merge" at all?

If not, should that be listed under the scenarios where this endpoint wouldn't work? Or is this scenario considered too irrelevant for nixpkgs?

Reply:
> Is there a branch-specific setting via rulesets for this?

I thought there was, but I can't find it either.



> Thus, I didn't mention that it doesn't work when "auto merge" is entirely disabled.

I think that's fine, as this tool deals exclusively with nixpkgs and the Auto Merge setting is unlikely to change. 🤞🏻 

I was more asking out of curiosity, as your wording made me wonder whether this endpoint is usable in other repos/scenarios, even when Auto Merge is not allowed.